File: BOJ/Kotlin/그리디 알고리즘/2217.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val N = readLine().toInt()
    val arr = arrayOfNulls<Int>(N)
    var res = 0

    for (i in 0 until N){
        arr[i] = readLine().toInt()
    }

    arr.sort()
    arr.reverse()

    for (i in 0 until N){
        if (arr[i]!! * (i+1) > res){
            res = arr[i]!! * (i+1)
        }
    }

    println(res)
}

File: BOJ/Python/그리디 알고리즘/2217.py
N = int(input())
arr = list()
res = 0

for _ in range(N):
    arr.append(int(input()))

arr.sort(reverse=True)

for count, rope in enumerate(arr):
    if rope * (count + 1) > res:
        res = rope * (count + 1)

print(res)